identity-account/src/account/builder.rs

```diff
@@ -33,7 +33,7 @@ use super::config::AutoSave;
 pub enum AccountStorage {
   Memory,
   #[cfg(feature = "stronghold")]
-  Stronghold(PathBuf, Option<String>),
+  Stronghold(PathBuf, Option<String>, Option<bool>),
```

Comment:
I'm a bit concerned that it's unclear to users what the `bool` flag is for. We could solve this in a few ways.

1. Add dedicated constructors to `AccountStorage`, e.g.

```rust
impl AccountStorage {
  pub const fn memory() -> AccountStorage {
    Self::Memory
  }

  pub fn stronghold(path: PathBuf, password: Option<String>) -> AccountStorage {
    Self::Stronghold(path, password)
  }

  pub fn custom(storage: Arc<dyn Storage>) -> AccountStorage {
    Self::Custom(storage)
  }
}
```

Now the question is, should we add `dropsave` to `AccountStorage::stronghold` and the enum variant? Probably not, to be ergonomic. If someone wants to disable `dropsave`, which should be the exception, then they can create their own `Stronghold` and use `AccountStorage::Custom`.

2. Don't add the flag at all in `AccountStorage` and use the default when creating the `Stronghold`. Same workaround as in 1 to set the option.
3. Get rid of `AccountStorage` entirely. I don't really see what benefit that enum brings, other than simplifying the build process slightly. Why have two ways of constructing a type. Especially when introducing options, constructing the type directly is more flexible than an enum variant, which we can't easily change breakingly to add more options. Without that type, we could just take an `impl Storage` (or `Arc<dyn Storage>` not sure) in the builder. It's slightly more effort to construct your own `Stronghold::new(&snapshot, passref).await?` but it's as ergonomic as `AccountStorage`. Same for `MemStore`. Setting the dropsave could just be `stronghold.set_dropsave(false)`, which is fine given how rare it probably is. Exposing the type more prominently like that means it's easier to add more setters for options in the future. Even bindings would benefit, because in Wasm only C-style enums are exposable, so we'd have to find a different solution anyway.

I would personally prefer 2 for this PR, or nothing at all if we decide to get rid of `AccountStorage` entirely (3) in a future PR. Any thoughts?

Reply:
I agree with the third option. I think we can get rid of the enum without adding complexity.

Reply:
I think it's unergonomic to set in the constructor every time, e.g. `Stronghold::new("./strong.hodl", "password", true).await?` because almost always you want it to be `true`? What if we add more options in the future? We cannot break the constructor. I think we should choose a solution that is extensible. I'm also fine with

```rust
// Can be extended in the future
pub struct StrongholdOptions {
   dropsave: bool,
}

impl StrongholdOptions {
    // acts like a builder
    fn dropsave(self, value: bool) -> Self { ... }
}

impl Stronghold {
    pub async fn new<'a, T, U>(snapshot: &T, password: U) -> Result<Self>;
    pub async fn new_with_options<'a, T, U>(snapshot: &T, password: U, options: StrongholdOptions) -> Result<Self>;
}
```

We used the `new_with_options` in the past in `IotaDocument` (though slightly differently) and I think that was a good solution.

Reply:
I really liked that